### ISSUE TITLE
Why does the VideoLLaMA3 demo webpage test exhibit output errors in the time granularity of video appearance?

### ISSUE THREAD

**Issue by LiangMeng89** [DAMO-NLP-SG/VideoLLaMA3#30]
Thank you very much for your open-source contribution, but when I tested the second dog.mp4 video on the website https://huggingface.co/spaces/lixin4ever/VideoLLaMA3, I encountered significant output errors in the time granularity of video appearance. Could you please tell me what might be causing this issue and how to resolve it?

This is my conversation test:
Q:
1.How long is this video in seconds?
2. How many times do dog scenes appear in the video?
3. How many dogs appear each time, and from which seconds to which seconds do they appear, respectively?
A:
1. The video is 8 seconds long.
2. The dog scenes appear three times in the video.
3. Each time, there are four dogs. The first dog scene appears at second 0, the second dog scene appears at second 1, and the third dog scene appears at second 2.

However,the correct answers are: 
1. The video is 10 seconds long.
2. Dog scenes appear 3 times in the video.
3. From 2 to 3 seconds, there are 5 dogs; from 3 to 4 seconds, there are 4 dogs; and from 9 to 10 seconds, there are 2 dogs.

The demo performance differs from what is described in the paper. Then,I will deploy VideoLLaMA3 locally to further evaluate its capability in understanding video granularity.

![Image](https://github.com/user-attachments/assets/db625587-6181-4e06-9072-c33b2bdfcb82)

**Comment by LiangMeng89** [comment]
This is the inference result of offline running on A800 (80G):

Loading checkpoint shards: 100%|█████████████████████████████████████| 4/4 [00:11<00:00,  2.85s/it]
1. The video lasts for 8 seconds.

2. Dog scenes appear three times in the video.

3. The first time, from 0 to 1.4 seconds, four dogs wearing astronaut suits appear inside a spacecraft. The second time, from 2.1 to 5.1 seconds, three dogs wearing astronaut suits are shown inside another spacecraft. The third time, from 6.9 to 8.7 seconds, two dogs wearing astronaut suits are seen inside yet another spacecraft.


Inference results show that the time accuracy in the video is still not high enough, slightly better than the online demo, adding the dog's dress and the description of the space inside and outside the aircraft.
Only the second answer is correct.
Take two seconds off the first answer.
The time period of the third answer was close to the true value range, only the third time the number of dogs was correct. The first time the 5 dogs appeared outside the spaceship, it was not described correctly.

**Comment by LiangMeng89** [comment]
Thanks for sharing your open source! @lixin4ever @hangzhang-nlp @yumingj @HZQ950419 

1.I'm doing event time inference on long videos close to 2min, and I still get time inaccuracy, is the sampling frequency you used to train videollama3 (less than 3min video: 1fps; more than 3min video: 180f. https://github.com/DAMO-NLP-SG/ VideoLLaMA3/issues/4)  too small?
2. If this is the reason, can I adjust the 1fps up to 10fps or higher on the A800 80G server for more accurate training or fine-tuning to extract key events and time information in my own video?
3. Can you share the dataset and specific calibration process for grounding and time calibration of the video?
4. What is the appropriate length of video in the dataset for training or fine-tuning?

Looking forward to your replies! Thank you!

<img width="1197" alt="Image" src="https://github.com/user-attachments/assets/38374ddf-957e-48c2-9a1b-8119e12d148a" />

**Comment by LiangMeng89** [comment]
> Thanks for sharing your open source! [@lixin4ever](https://github.com/lixin4ever) [@hangzhang-nlp](https://github.com/hangzhang-nlp) [@yumingj](https://github.com/yumingj) [@HZQ950419](https://github.com/HZQ950419)
> 
> 1.I'm doing event time inference on long videos close to 2min, and I still get time inaccuracy, , not reaching the timestamp extraction accuracy of an event depicted in the picture in the paper, is the sampling frequency you used to train videollama3 (less than 3min video: 1fps; more than 3min video: 180f. [https://github.com/[/issues/4](https://github.com/DAMO-NLP-SG/VideoLLaMA3/issues/4)#issuecomment-2628993541]) too small? 2. If this is the reason, can I adjust the 1fps up to 10fps or higher on the A800 80G server for more accurate training or fine-tuning to extract key events and time information in my own video? 3. Can you share the dataset and specific calibration process for grounding and time calibration of the video? 4. What is the appropriate length of video in the dataset for training or fine-tuning?
> 
> Looking forward to your replies! Thank you!
> 
> <img alt="Image" width="1197" src="https://private-user-images.githubusercontent.com/132972840/416219027-38374ddf-957e-48c2-9a1b-8119e12d148a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE3OTIwMjcsIm5iZiI6MTc0MTc5MTcyNywicGF0aCI6Ii8xMzI5NzI4NDAvNDE2MjE5MDI3LTM4Mzc0ZGRmLTk1N2UtNDhjMi05YTFiLTgxMTllMTJkMTQ4YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMxMlQxNTAyMDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03ZWFlOTFmNjJiODVhMTY1MjE5ZTIyZWQ1NzE5ZTQ1MDMwOTQzMjJkNzFmMDFlYjlmYjNkMjM5Mjk3YmFhZjk2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.W2KT4Eg8XUnrjdD4Cb4-OgBysuaVpFgToq0oSfN1FBI">

Could you upload this example video for us to test? @lixin4ever @hangzhang-nlp @yumingj @HZQ950419
